File: python/MainWindow.py
import sys
from TitleBar import *

class MainWindow(QMainWindow):
    def __init__(self):
        super(MainWindow, self).__init__()
        self.InitializeWindow()

    def InitializeWindow(self):
        self.setWindowFlags(Qt.FramelessWindowHint)
        self.setMinimumSize(WINDOW_DEFAULT_WIDTH, WINDOW_DEFAULT_HEIGHT);
        self.InitializeViews()
        self.setStyleSheet(str(self.LoadStyleFromQss(WINDOW_QSS)))

    def InitializeViews(self):
        self.titleBar = TitleBar(self)
        self.client = QWidget(self)
        self.center = QWidget(self)

        self.setCentralWidget(self.center)

        self.lay = QVBoxLayout(self)
        self.center.setLayout(self.lay)

        self.lay.addWidget(self.titleBar);
        self.lay.addWidget(self.client);
        self.lay.setStretch(1, 100);
        self.lay.setSpacing(0);
        self.lay.setContentsMargins(0, 0, 0, 0)

        self.titleBar.SetIcon(QPixmap(WINDOW_ICON));
        self.titleBar.SetTitle(WINDOW_TITLE);

    def LoadStyleFromQss(self, f):
        file = open(f)
        lines = file.readlines()
        file.close()
        res = ''
        for line in lines:
            res += line

        return res


if __name__ == '__main__':
    app = QApplication(sys.argv)

    win = MainWindow()
    win.show()

    sys.exit(app.exec_())
    pass

File: cpp/CustomTitleBar/MainWindow.cpp
#include "MainWindow.h"

MainWindow::MainWindow(QWidget *parent)
	: QMainWindow(parent)
{
	InitializeWindow();
}

void MainWindow::InitializeWindow()
{
	setWindowFlags(Qt::FramelessWindowHint);
	setMinimumSize(WIN_DEFAULT_WIDTH, WIN_DEFAULT_HEIGHT);
	InitializeViews();
	setStyleSheet(LoadStyleFromQSS(WINDOW_QSS));
}

void MainWindow::InitializeViews()
{
	titleBar = new TitleBar(this);
	clientWidget = new QWidget(this);
	centerWidget = new QWidget(this);
	setCentralWidget(centerWidget);

	lay = new QVBoxLayout(this);
	centerWidget->setLayout(lay);

	lay->addWidget(titleBar);
	lay->addWidget(clientWidget);
	lay->setStretch(1, 100);
	lay->setSpacing(0);
	lay->setMargin(0);

	titleBar->SetIcon(QPixmap(WINDOW_ICON));
	titleBar->SetTitle(WINDOW_TITLE);
}

QString MainWindow::LoadStyleFromQSS(const QString &str)
{
	QString line;
	try
	{
		QFile file(str);
		file.open(QFile::ReadOnly);

		line = file.readAll();
		file.close();
	}
	catch (const QException& e)
	{
		qDebug() << __FUNCTION__ << __LINE__ << e.what();
		return line;
	}

	return line;
}
